### PR TITLE
Add link checker to build, and disable hrefs for inherited methods/properties/operators

### PR DESCRIPTION
**Pull request by sethladd**

### CODE REVIEW

**Comment by sethladd** [comment]
Addresses https://github.com/dart-lang/dartdoc/issues/749

File: tool/grind.dart [hunk 1]
```diff
@@ -12,6 +12,8 @@ import 'package:grinder/grinder.dart';
 import 'package:librato/librato.dart';
 import 'package:path/path.dart' as path;
 import 'package:yaml/yaml.dart' as yaml;
+import 'package:html/parser.dart' show parse;
```
Comment (devoncarew) [inline tool/grind.dart]: nit: import sorting

File: lib/src/model.dart [hunk 2]
```diff
@@ -268,6 +268,7 @@ abstract class ModelElement implements Comparable {
   }
 
   String get href => package.isDocumented(this) ? _href : null;
+  //String get href => _href;
```
Comment (devoncarew) [inline lib/src/model.dart]: Is this still in progress?

File: tool/grind.dart [hunk 3]
```diff
@@ -274,3 +276,50 @@ Future _uploadStats(int sdkDocsGenTime) async {
     return librato.createAnnotation('build_ui', annotation);
   }
 }
+
+// TODO: check http links, check links in <link>
+@Task('Check links')
+checkLinks() {
```
Comment (devoncarew) [inline tool/grind.dart]: woot!

File: tool/grind.dart [hunk 4]
```diff
@@ -274,3 +276,50 @@ Future _uploadStats(int sdkDocsGenTime) async {
...
+        var normalized = path.normalize(full);
+        //print("    => $full\n      => $normalized");
+        doCheck(normalized, pathToCheck);
+      }
+    });
+  }
+
+  doCheck(start);
+
+  if (foundError) exit(1);
```
Comment (devoncarew) [inline tool/grind.dart]: `fail('unresolved links')`?

File: tool/grind.dart (hunk 1 above)
Comment (sethladd) [inline tool/grind.dart]: can haz analyzer quick fix? :P

File: tool/grind.dart (hunk 4 above)
Comment (sethladd) [inline tool/grind.dart]: Dumping a whole exception here is probably not that useful for the user. Did we ever add a failWithNoException ?

File: lib/src/model.dart (hunk 2 above)
Comment (sethladd) [inline lib/src/model.dart]: yup!

File: tool/grind.dart (hunk 4 above)
Comment (devoncarew) [inline tool/grind.dart]: Ah, forgot about that. PR's welcome :)

**Comment by sethladd** [comment]
Also addresses https://github.com/dart-lang/dartdoc/issues/751

**Comment by sethladd** [comment]
Also addresses https://github.com/dart-lang/dartdoc/issues/752

**Comment by sethladd** [comment]
OK @devoncarew please take a look.

This is a doosy. It closes out numerous issues, adds more tests, adds a link checker.

**Comment by devoncarew** [comment]
lgtm!

> Does it make sense to add the link checker to the Windows build bot?

Could be good, esp.if there are path issues we'd only see there.

**Comment by sethladd** [comment]
> Could be good, esp.if there are path issues we'd only see there.

OK, will do!  (looks like the Windows bot is driven off of grinder)

**Pull request closed without merging**